src/shared/utils/__tests__/guard.test.ts

```diff
@@ -0,0 +1,309 @@
+          "https://www.w3.org/2018/credentials/examples/v1",
+          "https://schemata.openattestation.com/com/openattestation/1.0/OpenAttestation.v3.json"
+        ],
+        issuer: {
+          name: "name",
+          id: "https://example.com"
+        },
+        issuanceDate: "2010-01-01T19:23:24Z",
+        type: ["VerifiableCredential", "UniversityDegreeCredential"],
+        credentialSubject: {
+          id: "did:example:ebfeb1f712ebc6f1c276e12ec21",
+          degree: {
+            type: "BachelorDegree",
+            name: "Bachelor of Science and Arts"
+          }
+        },
+        openAttestationMetadata: {
+          proof: {
+            value: "0xabcf",
+            type: v3.ProofType.OpenAttestationProofMethod,
+            method: v3.Method.DocumentStore
+          },
+          identityProof: {
+            identifier: "whatever",
+            type: v2.IdentityProofType.DNSTxt
+          }
+        },
+        name: "",
+        reference: "",
+        template: {
+          url: "https://",
+          name: "",
+          type: v3.TemplateType.EmbeddedRenderer
+        },
+        validFrom: "2010-01-01T19:23:24Z"
+      },
+      { version: SchemaId.v3 }
+    );
+    signedV2Document = await signDocument(wrappedV2Document, SUPPORTED_SIGNING_ALGORITHM.Secp256k1VerificationKey2018, {
+      public: "did:ethr:0xE712878f6E8d5d4F9e87E10DA604F9cB564C9a89#controller",
+      private: "0x497c85ed89f1874ba37532d1e33519aba15bd533cdcb90774cc497bfe3cde655"
+    });
+    signedV3Document = await signDocument(wrappedV3Document, SUPPORTED_SIGNING_ALGORITHM.Secp256k1VerificationKey2018, {
+      public: "did:ethr:0xE712878f6E8d5d4F9e87E10DA604F9cB564C9a89#controller",
+      private: "0x497c85ed89f1874ba37532d1e33519aba15bd533cdcb90774cc497bfe3cde655"
+    });
+  });
+  describe("isWrappedV2Document", () => {
+    // tests are a bit redundant with schema.test.ts
+    test("should be valid", () => {
+      expect(isWrappedV2Document(wrappedV2Document)).toBe(true);
+    });
+    test("should be invalid when document is v3", () => {
+      expect(isWrappedV2Document(wrappedV3Document)).toBe(false);
+    });
+    test("should not be valid when document is an empty object", () => {
+      expect(isWrappedV2Document({})).toBe(false);
+    });
+    test("should not be valid when document is null", () => {
+      expect(isWrappedV2Document(null)).toBe(false);
+    });
+    test("should not be valid when document.data is missing", () => {
+      expect(isWrappedV2Document(omit(cloneDeep(wrappedV2Document), "data"))).toBe(false);
+    });
+    test("should not be valid when document.data is null", () => {
+      expect(isWrappedV2Document(set(cloneDeep(wrappedV2Document), "data", null))).toBe(false);
+    });
+    test("should not be valid when document.data.issuers is missing", () => {
+      expect(isWrappedV2Document(set(cloneDeep(wrappedV2Document), "data.issuers", null))).toBe(false);
+    });
```

Comment:
should be `...omit(cloneDeep(wrappedV2Document), "data.issuers")...` instead

Reply:
done